rust/actyx/swarm/harness/src/bin/query.rs

```diff
@@ -44,24 +30,28 @@ fn main() -> anyhow::Result<()> {
 
     swarm_harness::setup_env()?;
     swarm_harness::run_netsim(opts, move |mut sim| async move {
-        let api = Api::new(&mut sim, app_manifest)?;
+        let api = Api::new(&mut sim, AppManifest::default())?;
 
-        for machine in sim.machines() {
+        for (idx, machine) in sim.machines().iter().enumerate() {
+            tracing::error!("{}", idx);
             api.run(machine.id(), |api| async move {
-                api.publish(PublishRequest { data: make_events(N) }).await?;
+                api.0
+                    .spawn_mut(|ax| {
```

Comment:
Curious, what does spawn_mut + block_on achieve compared to plainly calling `ax.publish()`?

Reply:
`ax` is behind `PinnedResource`:
```rust
/// Protects a resource so that it's always tied to a single, dedicated thread. This wrapper allows other
/// places to safely access the underlying resource even if that is not Sync or even Send. This
/// is achieved by having a separate, dedicated thread hosting the resource which then communicates
/// via a message queue with the rest of the system.
///
/// This wrapper is also useful to isolate a blocking resource from a thread-pool that hosts other
/// async tasks.
pub struct PinnedResource<T> {
    stopped: Arc<()>, // Just for keeping track of the count of external handles
    sender: Sender<Box<dyn FnOnce(&mut T) + Send>>,
}
```
From my interpretation, this is a way to keep accesses to the API orderly while changing machines and so on during the tests.
I've kept it as similar as possible to reduce entropy and to be able to deliver the PR, otherwise I'm guessing I'd still be investigating more things.

Reply:
IIRC this also ensures that the API calls are performed from a thread that is in the correct network namespace to talk to the Actyx instance.